### PR TITLE
chore(pkg): Prepare package for publication to jsr

### LINKED ISSUE #2073: Fix version parsing
```console
❯ deno run --reload -A https://deno.land/x/bids_validator@v1.14.8/bids-validator.ts --version
bids-validator v1.14.8
❯ deno run --reload -A https://raw.githubusercontent.com/bids-standard/bids-validator/v1.14.8/bids-validator/src/bids-validator.ts --version
bids-validator v1.14.8
❯ deno run --reload -A https://raw.githubusercontent.com/bids-standard/bids-validator/master/bids-validator/src/bids-validator.ts --version
bids-validator master
```

That said:

```console
❯ deno run --reload -A https://deno.land/x/bids_validator/bids-validator.ts --version
Warning Implicitly using latest version (v1.14.9-dev.0) for https://deno.land/x/bids_validator/bids-validator.ts
error: Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'split')
    return url.pathname.split('@')[1].split('/')[0]
                                     ^
    at getRemoteVersion (https://deno.land/x/bids_validator@v1.14.9-dev.0/version.ts:50:38)
    at getVersion (https://deno.land/x/bids_validator@v1.14.9-dev.0/version.ts:30:15)
    at parseOptions (https://deno.land/x/bids_validator@v1.14.9-dev.0/setup/options.ts:51:25)
    at main (https://deno.land/x/bids_validator@v1.14.9-dev.0/main.ts:11:25)
    at https://deno.land/x/bids_validator@v1.14.9-dev.0/bids-validator.ts:3:22
```

_Originally posted by @effigies in https://github.com/bids-standard/bids-validator/issues/1994#issuecomment-2272472975_

### PR DESCRIPTION
**Pull request by effigies**
Started playing with `deno publish --dry-run`.

1) Need package metadata to build.
2) Need to resolve type errors that are not caught by `deno test`.
3) Our dependencies need to be updated to use `npm:` or `jsr:`.
4) Need to get rid of symlinks and long filenames; I just excluded pretty much everything but `src/`.

With `deno.json` needing to contain a static version, falling back to parsing URLs doesn't make much sense instead of just importing from` deno.json`.

We'll need to make sure `deno.json` gets updated along with `lerna.json` during tags.

Open issue: Importing from a remote JSON file is incompatible with JSR releases. I'm not sure the best way to bundle a schema at build-time. We could start pushing `schema.json` and `metaschema.json` to a `schema` branch on the spec and include that as a submodule.

Closes #2073.

### CODE REVIEW

**Comment by nellh** [comment]
> Open issue: Importing from a remote JSON file is incompatible with JSR releases. I'm not sure the best way to bundle a schema at build-time. We could start pushing `schema.json` and `metaschema.json` to a `schema` branch on the spec and include that as a submodule.

Maybe we should publish the schema JSON as its own JSR package from that repo? That would be helpful for OpenNeuro to use it directly as well.

**Comment by effigies** [comment]
That was another thought I had. It seems reasonable to require a schema release of the spec in order to be the default schema for the validator. Bleeding edge users can always set the `BIDS_SCHEMA` environment variable.

**Comment by effigies** [comment]
Created https://jsr.io/@bids/schema

**Comment by effigies** [comment]
Current results of `deno publish --dry-run`:

```console
❯ deno publish --dry-run
Checking for slow types in the public API...
warning[unanalyzable-dynamic-import]: unable to analyze dynamic import
  --> /home/chris/Projects/bids/validator/bids-validator/src/setup/loadSchema.ts:22:58
   | 
22 |     const schemaModule = await import(/* @vite-ignore */ schemaUrl, {
   |                                                          ^^^^^^^^^ the unanalyzable dynamic import
   | 

  info: after publishing this package, imports from the local import map / package.json do not work
  info: dynamic imports that can not be analyzed at publish time will not be rewritten automatically
  info: make sure the dynamic import is resolvable at runtime without an import map / package.json

Simulating publish of @bids/validator@1.14.9-dev.0 with files:
   file:///home/chris/Projects/bids/validator/LICENSE (1.1KB)
   file:///home/chris/Projects/bids/validator/bids-validator/README.md (35.04KB)
   file:///home/chris/Projects/bids/validator/bids-validator/deno.json (1.13KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/.git-meta.json (148B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/README.md (3.05KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/bids-validator.ts (155B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/browser.test.ts (3.28KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/browser.ts (2.54KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/deno.test.ts (3.23KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/deno.ts (4.16KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/dwi.ts (580B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/ignore.test.ts (864B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/ignore.ts (1.01KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/inheritance.ts (1.51KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/json.test.ts (2KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/json.ts (971B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/nifti.test.ts (1.95KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/nifti.ts (2.16KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/files/tsv.ts (1.36KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/issues/datasetIssues.test.ts (2.3KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/issues/datasetIssues.ts (2.17KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/issues/list.test.ts (1.45KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/issues/list.ts (6.21KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/main.ts (1.39KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/applyRules.test.ts (6.99KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/applyRules.ts (16.65KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/associations.ts (4.84KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/context.test.ts (1.43KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/context.ts (7.89KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/entities.test.ts (3.21KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/entities.ts (820B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/expressionLanguage.test.ts (7.83KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/expressionLanguage.ts (2.78KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/fixtures.test.ts (2.64KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/modalities.ts (497B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/walk.test.ts (1.17KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/schema/walk.ts (1.58KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/setup/loadSchema.test.ts (1.1KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/setup/loadSchema.ts (1.39KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/setup/options.test.ts (427B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/setup/options.ts (2.02KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/setup/requestPermissions.ts (617B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/summary/collectSubjectMetadata.ts (2.1KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/summary/summary.test.ts (600B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/summary/summary.ts (4.46KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/README.md (30B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/bom-utf16.tsv (14B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/bom-utf8.json (42B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/generate-filenames.ts (762B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/local/bids_examples.test.ts (2.37KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/local/common.ts (1.31KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/local/empty_files.test.ts (1.41KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/local/hed-integration.test.ts (2.62KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/local/valid_dataset.test.ts (616B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/local/valid_filenames.test.ts (614B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/local/valid_headers.test.ts (1.25KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/nullReadBytes.ts (109B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/schema-expression-language.test.ts (1.96KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/tests/simple-dataset.ts (1.71KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/types/check.ts (659B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/types/columns.test.ts (1.48KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/types/columns.ts (1.4KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/types/context.ts (2.75KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/types/filetree.ts (1.6KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/types/issues.ts (1.09KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/types/schema.ts (1.9KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/types/validation-result.ts (797B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/utils/errors.ts (235B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/utils/logger.test.ts (1.38KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/utils/logger.ts (1.05KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/utils/memoize.ts (996B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/utils/objectPathHandler.ts (609B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/utils/output.ts (4.42KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/bids.test.ts (0B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/bids.ts (4.07KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/filenameIdentify.test.ts (3.24KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/filenameIdentify.ts (5.99KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/filenameValidate.test.ts (1.79KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/filenameValidate.ts (7.33KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/hed.ts (3.05KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/internal/emptyFile.ts (333B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/internal/unusedFile.ts (1.35KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/isBidsy.ts (548B)
   file:///home/chris/Projects/bids/validator/bids-validator/src/validators/json.ts (1.06KB)
   file:///home/chris/Projects/bids/validator/bids-validator/src/version.ts (1.59KB)
Warning Aborting due to --dry-run
```

File: bids-validator/deno.json [hunk 1]
```diff
@@ -1,4 +1,39 @@
...
+  },
+  "imports": {
+    "@std/assert": "jsr:@std/assert@1.0.2",
+    "@std/fmt": "jsr:@std/fmt@1.0.0",
+    "@std/fs": "jsr:@std/fs@1.0.1",
+    "@std/io": "jsr:@std/io@0.224.4",
+    "@std/log": "jsr:@std/log@0.224.5",
+    "@std/path": "jsr:@std/path@1.0.2",
+    "@ajv": "npm:ajv@8.16.0",
+    "@bids/schema": "jsr:@bids/schema@0.0.20240813",
```
Comment (effigies) [inline bids-validator/deno.json]:
```suggestion
    "@bids/schema": "jsr:@bids/schema@0.0.202408141737",
```

File: bids-validator/src/setup/loadSchema.ts [hunk 2]
```diff
@@ -1,8 +1,6 @@
 import { Schema } from '../types/schema.ts'
 import { objectPathHandler } from '../utils/objectPathHandler.ts'
-import * as schemaDefault from 'https://bids-specification.readthedocs.io/en/latest/schema.json' assert {
-  type: 'json',
-}
+import * as schemaDefault from '@bids/schema' with { type: 'json' }
```
Comment (effigies) [inline bids-validator/src/setup/loadSchema.ts]:
```suggestion
import * as schemaDefault from '@bids/schema/schema' with { type: 'json' }
```

**Pull request merged**